File: leetcode/fun-with-arrays/max_consecutive_ones.js
// https://leetcode.com/explore/learn/card/fun-with-arrays/521/introduction/3238/

var findMaxConsecutiveOnes = function(nums) {
    let search = 0;
    let track = 0;

    for(let i = 0; i < nums.length; i++)
    {
        if(nums[i] = 1) {
            search++;
            if(search > track) {
                track = search;
            }
        } else {
            search = 0;
        }
    }
    return track;
};

File: leetcode/fun-with-arrays/max_consecutive_ones.py
def findMaxConsecutiveOnes(nums) -> int:
    search = 0
    count = 0

    for i in range(0, len(nums)):
        if nums[i] == 1:
            search +=1
            if search > count:
                count = search
        else:
            search = 0

    return count


nums = [1,1,0,1,1,1]
b = findMaxConsecutiveOnes(nums)

print(b)
